src/app/data-client.ts

```diff
@@ -709,6 +714,157 @@ export class DataClient {
     return datasets;
   }
 
+  /**
+   * Uploads the content and metadata for tabular data.
+   *
+   * Upload tabular data collected on a robot through a specific component
+   * (e.g., a motor) along with the relevant metadata to app.viam.com. Tabular
+   * data can be found under the "Sensors" subtab of the Data tab on
+   * app.viam.com.
+   *
+   * @param tabularData The list of data to be uploaded, represented tabularly
+   *   as an array.
+   * @param partId The part ID of the component used to capture the data
+   * @param componentType The type of the component used to capture the data
+   *   (e.g., "movementSensor")
+   * @param componentName The name of the component used to capture the data
+   * @param methodName The name of the method used to capture the data.
+   * @param tags The list of tags to allow for tag-based filtering when
+   *   retrieving data
+   * @param dataRequestTimes Array of Date arrays, each containing two `Date`
+   *   objects denoting the times this data was requested[0] by the robot and
```

Comment:
instead of an array of arrays, can we have this be an array of `[Datetime, Datetime]` tuples, so we don't invite the possibility of people including an incorrect amount of data?

Reply:
Ah, yes. I remember why I changed this from `[Date, Date][]` to an array of arrays. I was getting an error saying: `src/app/data-client.test.ts(1007,10): error TS2322: Type 'Date[]' is not assignable to type '[Date, Date]'.`.

I agree that having a tuple is the way to go. It does means that when someone is trying to fill in this value, they can't just do:
```
const var = [date1, date2]
await client.tabularDataCaptureUpload(..., var)
```
They now have to declare the type: `const var: [Date, Date] = [date1, date2]`. I think this is okay, because while it is more annoying, it does make it more purposeful.

Reply:
Oh, that was a listing error, so users can still just use the simple `const var = [date1, date2]`!